src/words/exceptions/WordsReferenceException.java

```diff
@@ -2,10 +2,17 @@
 @SuppressWarnings("serial")
 public class WordsReferenceException extends WordsRuntimeException {
 	
+	private String message;
+	
+	public WordsReferenceException(String message) {
+		this.message = message;
+	}
+	
 	public WordsReferenceException() {
+		this.message = "attempted to dereference nothing";
```

Comment:
I think this isn't quite the right description for how this exception is being used.  This exception is being used any time we try to dereference something that is not an object.  That includes NOTHING, but it also includes anything else too (number, string).